service/frontend/workflowHandler.go

```diff
@@ -3706,6 +3717,69 @@ func isFailoverRequest(updateRequest *gen.UpdateDomainRequest) bool {
 	return updateRequest.ReplicationConfiguration != nil && updateRequest.ReplicationConfiguration.ActiveClusterName != nil
 }
 
+func isGraceFailoverRequest(updateRequest *gen.UpdateDomainRequest) bool {
+	return updateRequest.IsSetFailoverTimeoutInSeconds()
+}
+
+func (wh *WorkflowHandler) checkOngoingFailover(
+	ctx context.Context,
+	domainName *string,
+) error {
+
+	clusterMetadata := wh.GetClusterMetadata()
+	respChan := make(chan *gen.DescribeDomainResponse)
+	wg := &sync.WaitGroup{}
+
+	decribeDomain := func(
+		ctx context.Context,
+		client frontend.Client,
+		domainName *string,
+	) {
+		defer wg.Done()
+		resp, _ := client.DescribeDomain(
+			ctx,
+			&gen.DescribeDomainRequest{
+				Name: domainName,
+			},
+		)
+		respChan <- resp
+	}
+
+	for clusterName, cluster := range clusterMetadata.GetAllClusterInfo() {
+		if !cluster.Enabled {
+			continue
+		}
+		frontendClient := wh.GetRemoteFrontendClient(clusterName)
+		wg.Add(1)
+		go decribeDomain(
+			ctx,
+			frontendClient,
+			domainName,
+		)
+	}
+	wg.Wait()
```

Comment:
`respChan` has 0 capacity and `wg.Done()` is called after resp has been sent. If `wg.Wait()` is called here, I think it will result in a deadlock because there's no receiver for the channel -> resp can't be sent -> `wg.Done()` won't be called.

I would suggest get rid of the wait group entirely. You already know how many responses will be sent when creating the go routine, just use a for loop to receive responses from `respChan`.

Reply:
Good catch. Set the channel size and still use the wait group.